pkg/kubelet/kubelet.go

```diff
@@ -1393,6 +1397,13 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 		klog.Fatal(err)
 	}
 
+	if kl.kubeClient != nil {
+		// start informer for CSIDriver
+		if utilfeature.DefaultFeatureGate.Enabled(features.CSIDriverRegistry) {
+			go kl.informerFactory.Start(wait.NeverStop)
```

Comment:
The layering is a little difficult to follow:
* The shared informer factory is created in kubelet volume host
* The informer is started in kubelet
* Wait for cache sync is in csi_util

Maybe for now, let's manage everything in the kubelet volume host?

Reply:
Made changes based on our discussion.